File: src/main/java/ru/scarlet/filestorage/repository/AttachmentRepository.java
package ru.scarlet.filestorage.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.scarlet.filestorage.entity.Attachment;

import java.util.List;
import java.util.UUID;

@Repository
public interface AttachmentRepository extends JpaRepository<Attachment, UUID> {
    List<Attachment> findByAuthorOrderByCreatedDesc(String author);
}

File: src/main/java/ru/scarlet/filestorage/repository/AttachmentRepository.kt
package ru.scarlet.filestorage.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.scarlet.filestorage.entity.Attachment
import java.util.*

@Repository
interface AttachmentRepository : JpaRepository<Attachment?, UUID?> {
    fun findByAuthorOrderByCreatedDesc(author: String?): List<Attachment?>?
}